### ISSUE TITLE
Unable to jump between find results in MS Word if using legacy object model

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#16915]
### Steps to reproduce:
1. Type "world, world, world" in MS Word.
2. NVDA+space switches to browse mode.
3. Ctrl+Home to the top of the document
4. NVDA+Ctrl+F opens the search dialog and enter "world".
5. Press Enter.
6. Press NVDA+F3
### Actual behavior:
NVDA cannot jump to the second find result, the caret is not updated.

### Expected behavior:
NVDA can jump between all matching results as if UIA was enabled
### NVDA logs, crash dumps and other attachments:
```log
IO - speech.speech.speak (11:35:01.809) - MainThread (6520):
Speaking ['日志片段开始点已标记，再按一次复制到剪贴板']
IO - inputCore.executeGesture (11:35:02.735) - winInputHook (4048):
Input: kb(laptop):NVDA+f3
IO - speech.speech.speak (11:35:02.858) - MainThread (6520):
Speaking ['World, world, world']
DEBUG - UIAHandler.shouldUseUIAInMSWord (11:35:02.863) - Dummy-12 (8744):
Not using UIA in MS Word on pre Windows 11 OS due to missing custom extensions
IO - inputCore.executeGesture (11:35:03.395) - winInputHook (4048):
Input: kb(laptop):NVDA+f3
IO - speech.speech.speak (11:35:03.586) - MainThread (6520):
Speaking ['World, world, world']
DEBUG - UIAHandler.shouldUseUIAInMSWord (11:35:03.595) - Dummy-12 (8744):
Not using UIA in MS Word on pre Windows 11 OS due to missing custom extensions
DEBUG - speech.manager.SpeechManager._handleIndex (11:35:03.595) - MainThread (6520):
Unknown index 3226, speech probably cancelled from main thread.
IO - inputCore.executeGesture (11:35:04.746) - winInputHook (4048):
Input: kb(laptop):control+shift+NVDA+f1
```
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2024.3Beta4
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.4651
#### Name and version of other software in use when reproducing the issue:
M365 MSO (Version 2406 Build 16.0.17726.20078) 64 Bit 
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by cary-rowen** [comment]
I was surprised. I thought such a Highly used feature should have been reported earlier, but I didn't search for similar issues.
Thanks to @CyrilleB79 for assisting with more information, which let me know it's not a regression for a recent update.

**Comment by Adriani90** [comment]
Given the accessible and quite advanced search feature in MS Word, imho this is not of a high priority. What is the advantage of NVDA‘s find feature over the native feature in MS Word? You can move between search results by pressing control plus page up page down

**Comment by cary-rowen** [comment]
Note that this is a bug in the first place and I don't think you can define its priority either.
It's part of NVDA's browsing mode, which may be confusing to users who are used to this feature in web browsing mode.

For Windows 10 users, UIA is not enabled by default, which makes the situation even worse.

**Comment by CyrilleB79** [comment]
This issue also impacts Outlook.
In Outlook, messages being read are in browse mode by default and NVDA's search feature would be the most appropriate if it worked.

I do not know how the native find feature works in Outlook.
You can use `control+H` (search and replace), "find next" button (or its shortcut) and tab back and forth with control+tab between the document and the dialog to press "Next".
But it's not very user friendly.

Better use NVDA's find feature and press `leftArrow` before `NVDA+f3` to workaround the current issue.

**Comment by britechguy** [comment]
> What is the advantage of NVDA‘s find feature over the native feature in MS Word?

In my opinion, it's twofold:
1) Getting users to employ what I think is a grossly underused screen reader feature.
2) For simple consistency.

If NVDA find is supposed to be a text search function, it should be able to search for and locate all occurrences of a given piece of text in any program where text is what's being worked with.  And nowhere is text being worked with more than MS-Word.

I discovered this issue, strangely enough, when doing a "compare and contrast" after I had an issue with JAWS Find of a slightly different nature.  This issue is equally annoying, as I expect a find to, well, find across contexts if the search string is present, and if the "whole word" option is not chosen, everywhere it occurs, including embedded within other words.
